src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/RequestHandler.java

```diff
@@ -122,86 +121,75 @@ public static void assignRequest(final IStandardRequestManager manager, final IR
 
         request.setState(new WrappedStaticStateRequestManager(manager), RequestState.ASSIGNING);
 
-        @SuppressWarnings(RAWTYPES) final Set<TypeToken> requestTypes = ReflectionUtils.getSuperClasses(request.getRequestType());
+        final Set<TypeToken> requestTypes = ReflectionUtils.getSuperClasses(request.getRequestType());
         requestTypes.remove(TypeConstants.OBJECT);
 
-        final Set<IToken<?>> failedResolvers = new HashSet<>();
+        final List<TypeToken> typeIndexList = new LinkedList<>(requestTypes);
 
-        for (@SuppressWarnings(RAWTYPES) final TypeToken requestType : requestTypes)
+        final Set<IRequestResolver<?>> resolvers = requestTypes.stream()
+                .filter(typeToken -> manager.getRequestableTypeRequestResolverAssignmentDataStore().getAssignments().containsKey(typeToken))
+                .flatMap(type -> manager.getRequestableTypeRequestResolverAssignmentDataStore()
+                        .getAssignments()
+                        .get(type)
+                        .stream()
+                        .map(iToken -> ResolverHandler.getResolver(manager, iToken)))
+                .filter(iRequestResolver -> typeIndexList.contains(iRequestResolver.getRequestType()))
+                .sorted(Comparator.comparingInt((IRequestResolver<?> r) -> -1 * r.getPriority())
+                        .thenComparingInt((IRequestResolver<?> r) -> typeIndexList.indexOf(r.getRequestType())))
+                .collect(Collectors.toCollection(LinkedHashSet::new));
+
+
+        for (final IRequestResolver<?> resolver : resolvers)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS135)